eZ/Publish/Core/FieldType/Validator/DestinationContentValidator.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\FieldType\Validator;
+
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use eZ\Publish\API\Repository\Exceptions\UnauthorizedException;
+use eZ\Publish\Core\FieldType\ValidationError;
+
+/**
+ * Validator for checking existence of destination content.
+ */
+class DestinationContentValidator
+{
+    /** @var \eZ\Publish\API\Repository\ContentService */
+    private $contentService;
+
+    public function __construct(?ContentService $contentService = null)
+    {
+        $this->contentService = $contentService;
+    }
+
+    public function validate($value): ?ValidationError
```

Comment:
What is this $value. What type it has? Can it be more specific than "value"?

Reply:
It can be anything I guess. If you use the Public PHP API we can put really anything there, that's why I have phpdoc'ed it as `mixed`.